src/runtime/str.cpp

```diff
@@ -1146,9 +1146,24 @@ extern "C" Box* strMul(BoxedString* lhs, Box* rhs) {
     assert(PyString_Check(lhs));
 
     int n;
+
+    if (PyIndex_Check(rhs)) {
+        Box* index = PyNumber_Index(rhs);
+        if (PyErr_Occurred()) {
```

Comment:
minor, but PyErr_Occurred is slower than you'd expect and it's good to check the return value instead.  (Doesn't seem like it really matters here but it's good practice.)